File: cpp/Process.cpp
#include <iostream>
#include <vector>
#include <string>
#include <iomanip>

using namespace std;

class Human
{
private:
    // deklar variabel class human
    string nik, name, jenis_kelamin;

public:
    // constructor  human
    Human(string nik, string name, string jenis_kelamin)
    {
        this->nik = nik;
        this->name = name;
        this->jenis_kelamin = jenis_kelamin;
    }
    // getter/setter methods untuk class human
    string get_nik()
    {
        return nik;
    }

    void set_nik(string nik)
    {
        this->nik = nik;
    }

    string get_name()
    {
        return name;
    }

    void set_name(string name)
    {
        this->name = name;
    }

    string get_jenis_kelamin()
    {
        return jenis_kelamin;
    }

    void set_jenis_kelamin(string jenis_kelamin)
    {
        this->jenis_kelamin = jenis_kelamin;
    }
};

class Temp : public Human
{
private:
    // deklar variabel class Temp atau mahasiswa
    string nim, prodi, fakultas;

public:
    // constructor temp atau mahasiwa
    Temp(string nik, string name, string jenis_kelamin, string nim, string prodi, string fakultas) : Human(nik, name, jenis_kelamin) //: adalah super dari CPP
    {
        this->nim = nim;
        this->prodi = prodi;
        this->fakultas = fakultas;
    }

    // getter/setter methods untuk class temp atau mahasiswa
    string get_nim()
    {
        return nim;
    }

    void set_nim(string nim)
    {
        this->nim = nim;
    }

    string get_prodi()
    {
        return prodi;
    }

    void set_prodi(string prodi)
    {
        this->prodi = prodi;
    }

    string get_fakultas()
    {
        return fakultas;
    }

    void set_fakultas(string fakultas)
    {
        this->fakultas = fakultas;
    }
};

class SivitasAkademik : public Temp
{
private:
    // deklar variabel class sivitas akademik
    string asal_univ, email;

public:
    // constructor sivitasAkademik
    SivitasAkademik(string nik, string name, string jenis_kelamin, string nim, string prodi, string fakultas, string asal_univ, string email) : Temp(nik, name, jenis_kelamin, nim, prodi, fakultas)
    {
        this->asal_univ = asal_univ;
        this->email = email;
    }
    // getter/setter methods untuk class Sivitas akademik
    string get_asal_univ()
    {
        return asal_univ;
    }

    void set_asal_univ(string asal_univ)
    {
        this->asal_univ = asal_univ;
    }

    string get_email()
    {
        return email;
    }

    void set_email(string email)
    {
        this->email = email;
    }
};

File: python/Process.py
class Human:
    def __init__(self, nik, name, jenis_kelamin):  # constructed human
        self.nik = nik
        self.name = name
        self.jenis_kelamin = jenis_kelamin

    # getter/setter methods untuk class human
    def get_nik(self):
        return self.nik

    def set_nik(self, nik):
        self.nik = nik

    def get_name(self):
        return self.name

    def set_name(self, name):
        self.name = name

    def get_jenis_kelamin(self):
        return self.jenis_kelamin

    def set_jenis_kelamin(self, jenis_kelamin):
        self.jenis_kelamin = jenis_kelamin


class Temp(Human):

    # constructor temp atau mahasiwa
    def __init__(self, nik, name, jenis_kelamin, nim, prodi, fakultas):
        # pemanggilan variabel di kelas parent
        super().__init__(nik, name, jenis_kelamin)
        self.nim = nim
        self.prodi = prodi
        self.fakultas = fakultas

    # getter/setter methods untuk class temp atau mahasiswa
    def get_nim(self):
        return self.nim

    def set_nim(self, nim):
        self.nim = nim

    def get_prodi(self):
        return self.prodi

    def set_prodi(self, prodi):
        self.prodi = prodi

    def get_fakultas(self):
        return self.fakultas

    def set_fakultas(self, fakultas):
        self.fakultas = fakultas


class SivitasAkademik(Temp):
    # constructor sivitasAkademik
    def __init__(self, nik, name, jenis_kelamin, nim, prodi, fakultas, asal_univ, email):
        # pemanggilan variabel kelas parent
        super().__init__(nik, name, jenis_kelamin, nim, prodi, fakultas)
        self.asal_univ = asal_univ
        self.email = email

    # getter/setter methods untuk class Sivitas akademik
    def get_asal_univ(self):
        return self.asal_univ

    def set_asal_univ(self, asal_univ):
        self.asal_univ = asal_univ

    def get_email(self):
        return self.email

    def set_email(self, email):
        self.email = email
